### ISSUE TITLE
Report empty 'List<T>' creations if 'IReadOnlyList<T>' is returned

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#495]
If a method or a property returns an `IReadOnlyList<T>` and there is a
```c#
return new List<T>();
```
call inside, we should report an issue and propose to fix it with
```c#
return Array.Empty<T>();
```
